homeassistant/components/aussie_broadband/sensor.py

```diff
@@ -42,60 +43,75 @@ async def async_update_data():
-            CounterEntity(
-                coordinator, service_id, "Billing Cycle Length", "daysTotal", "days"
-            ),
-            CounterEntity(
-                coordinator,
-                service_id,
-                "Billing Cycle Remaining",
-                "daysRemaining",
-                "days",
-            ),
+            AussieBroadandTotalUsage(coordinator, service_id),
+            AussieBroadandDownloaded(coordinator, service_id),
+            AussieBroadandUploaded(coordinator, service_id),
+            AussieBroadandBillingCycleLength(coordinator, service_id),
+            AussieBroadandBillingCycleRemaining(coordinator, service_id),
         ]
     )
 
     return True
 
 
-class CounterEntity(CoordinatorEntity, SensorEntity):
-    """Representation of a Aussie Broadband counter metric sensor."""
+class AussieBroadandSensorEntity(CoordinatorEntity, SensorEntity):
+    """Base class for Aussie Broadband metric sensors."""
+
+    _attribute: str
 
     def __init__(
         self,
         coordinator: DataUpdateCoordinator,
         service_id: int,
-        name: str,
-        attribute: str,
-        unit_of_measurement: str,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
-
-        self._service_id = service_id
-        self._name = name
-        self._attribute = attribute
-        self._unit_of_measurement = unit_of_measurement
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def unit_of_measurement(self):
-        """Return unit of measurement."""
-        return self._unit_of_measurement
+        self._attr_unique_id = f"{service_id}:{self._attribute}"
 
     @property
     def state(self):
         """Return the state of the device."""
         return self.coordinator.data[self._attribute]
 
-    @property
-    def unique_id(self) -> str | None:
-        """Return a unique ID."""
-        return str(self._service_id) + ":" + self._attribute
+
+class AussieBroadandTotalUsage(AussieBroadandSensorEntity):
+    """Representation of a Aussie Broadband Total Usage sensor."""
+
+    _attribute = "usedMb"
+    _attr_name = "Total Usage"
+    _attr_unit_of_measurement = DATA_MEGABYTES
+    _attr_state_class = STATE_CLASS_TOTAL_INCREASING
+
+
+class AussieBroadandDownloaded(AussieBroadandSensorEntity):
+    """Representation of a Aussie Broadband Download Usage sensor."""
+
+    _attribute = "downloadedMb"
+    _attr_name = "Downloaded"
+    _attr_unit_of_measurement = DATA_MEGABYTES
+    _attr_state_class = STATE_CLASS_TOTAL_INCREASING
+
+
+class AussieBroadandUploaded(AussieBroadandSensorEntity):
+    """Representation of a Aussie Broadband Upload Usage sensor."""
+
+    _attribute = "uploadedMb"
+    _attr_name = "Uploaded"
+    _attr_unit_of_measurement = DATA_MEGABYTES
+    _attr_state_class = STATE_CLASS_TOTAL_INCREASING
```

Comment:
Perhaps `_attr_last_reset` would be appropriate to pair with this? (and for the others)

Reply:
Not anymore, see https://developers.home-assistant.io/blog/2021/08/16/state_class_total/
last_reset is being removed already.